app/scripts/components/analysis/results/analysis-head-actions.tsx

```diff
@@ -65,7 +62,7 @@ const MetricList = styled.ul`
   }
 `;
 
-const MetricSwitch = styled(FormSwitch)`
+const MetricSwitch = styled(FormSwitch)<{ metricThemeColor: string }>`
```

Comment:
Below in line 74, we might be able to avoid casting to `any` - see https://medium.com/@KevinBGreene/fun-with-typescript-template-literal-types-95580e27e0e1

Reply:
Not sure it is in this case, and once again it has to do with not phenomenal types on the UI library.

The parameter for `themeVal(path)` must be a valid theme object path. Even though something like `color.infographicA` would be valid in this app's theme, the list of theme paths are manually [defined in the ui library](https://github.com/developmentseed/ui-library-seed/blob/5f00bf13e66c6afa7f46f22c46a76eaf9fa1e18a/packages/theme-provider/src/utils.d.ts#L8-L197).

I know that there are ways to [dynamically generate the paths](https://dev.to/bwca/create-object-property-string-path-generator-with-typescript-13e3) so could be something to explore.

Reply:
Nevertheless, let me know if you have any suggestion on how to approach it.